src/app/shared/main-layout/header/header.component.html

```diff
@@ -1,14 +1,14 @@
 <header>
     <app-logo></app-logo>
-    <div class="header__user-details">
+    <div class="header__user-details" *ngIf="checkStatus()">
```

Comment:
calling methods in template - is not a good idea as well
your method will be called frequently

Reply:
observables - the best solution

Reply:
fixed